packages/hawtio/src/plugins/shared/attributes/AttributeTable.tsx

```diff
@@ -5,34 +5,81 @@ import { PluginNodeSelectionContext } from '@hawtiosrc/plugins/selectionNodeCont
 
 export const AttributeTable: React.FunctionComponent = () => {
   const { selectedNode } = useContext(PluginNodeSelectionContext)
-  const [attributesList, setAttributesList] = useState<AttributeValues[]>([{}])
+  const [attributesList, setAttributesList] = useState<{ [name: string]: AttributeValues }>({})
   const [isReading, setIsReading] = useState<boolean>(false)
+  const attributesEntries = Object.values(attributesList)
+
+  async function readAttributes(currentSelection: MBeanNode) {
+    if (!currentSelection) return
+
+    const childrenMbeansAttributes: { [name: string]: AttributeValues } = {}
+
+    setIsReading(true)
+
+    for (const node of currentSelection.getChildren()) {
+      if (!node || !node?.objectName) continue
+
+      const attrs = await attributeService.read(node.objectName)
+      childrenMbeansAttributes[node.objectName] = attrs
+    }
+
+    setAttributesList({ ...childrenMbeansAttributes })
+
+    setIsReading(false)
+  }
+
+  async function setJobForSpecificNode(node: MBeanNode | null): Promise<void> {
+    if (!node || !node?.objectName) return
+
+    const mbean = node.objectName
+    attributeService.register({ type: 'read', mbean }, (response: IResponse) => {
+      attributesList[mbean] = response.value as AttributeValues
+      setAttributesList({ ...attributesList })
+    })
+  }
+
+  async function setReadingJobs(currentSelection: MBeanNode): Promise<void> {
+    if (!currentSelection) return
+
+    currentSelection.getChildren().forEach(async node => await setJobForSpecificNode(node))
+  }
+
+  function checkIfAllMBeansHaveSameAttributes(attributesEntries: AttributeValues[]): boolean {
+    if (attributesEntries.length <= 1) {
+      return true
+    }
+
+    const firstMBeanAttributesElements = attributesEntries[0].length
+
+    if (!attributesEntries.every(mbeanAttributes => mbeanAttributes.length === firstMBeanAttributesElements)) {
+      return false
+    }
+
+    const labelSet: Set<string> = new Set()
+    Object.keys(attributesEntries[0]).forEach(label => labelSet.add(label))
+
+    return attributesEntries.every(attributes => Object.keys(attributes).every(label => labelSet.has(label)))
+  }
 
   useEffect(() => {
     if (!selectedNode) {
       return
     }
 
-    const readAttributes = async () => {
-      const childrenMbeansAttributes: AttributeValues[] = []
-      setIsReading(true)
-      if (selectedNode.children)
-        for (const mbean of selectedNode.children) {
-          const objectName = mbean.objectName
-          if (objectName) {
-            const attrs = await attributeService.read(objectName)
-            childrenMbeansAttributes.push(attrs)
-          }
-        }
-      setAttributesList([...childrenMbeansAttributes])
-      setIsReading(false)
-    }
-    readAttributes()
+    readAttributes(selectedNode)
+    setReadingJobs(selectedNode)
+
+    return () => attributeService.unregisterAll()
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Needed if I don't want to put all the functions inside of the useEffect or have to use useCallback for every one (even though they don't have any dependencies with outside code)